src/python/pants/binaries/binary_util.py

```diff
@@ -237,29 +238,29 @@ def _fetch_binary(self, name, binary_path):
     bootstrap_dir = os.path.realpath(os.path.expanduser(self._pants_bootstrapdir))
     bootstrapped_binary_path = os.path.join(bootstrap_dir, binary_path)
     if not os.path.exists(bootstrapped_binary_path):
-      downloadpath = bootstrapped_binary_path + '~'
-      try:
+      safe_mkdir_for(bootstrapped_binary_path)
```

Comment:
I think you solved this already: `with safe_concurrent_creation(bootstrapped_binary_path) as downloadpath:`

Reply:
Right! I forgot about that somehow. Done.